### PR TITLE
feat: Security integrations datasource v1 readiness

### PR DESCRIPTION
**Pull request by sfc-gh-jmichalak**
Handle fields in describe for all types of security integrations.

## Test Plan

* [x] acceptance tests

### CODE REVIEW

File: pkg/schemas/security_integration.go [hunk 1]
```diff
@@ -1,63 +1,38 @@
...
-			"OAUTH_ALLOWED_AUTHORIZATION_ENDPOINTS",
-			"OAUTH_ALLOWED_TOKEN_ENDPOINTS":
-			securityIntegrationProperties[strings.ToLower(property.Name)] = []map[string]any{SecurityIntegrationPropertyToSchema(&property)}
-		default:
-			log.Printf("[WARN] unexpected property %v returned from Snowflake", property.Name)
+	for _, desc := range integrationProperties {
+		desc := desc
+		if slices.Contains(allSecurityIntegrationPropertiesNames, desc.Name) {
+			securityIntegrationProperties[strings.ToLower(desc.Name)] = []map[string]any{SecurityIntegrationPropertyToSchema(&desc)}
 		}
```
Comment (sfc-gh-jcieslak) [inline pkg/schemas/security_integration.go:36]: no else ?

File: pkg/schemas/oauth_security_integration_for_partner_applications.go [hunk 2]
```diff
@@ -30,33 +31,36 @@ var DescribeOauthIntegrationForPartnerApplications = map[string]*schema.Schema{
...
-			"OAUTH_AUTHORIZATION_ENDPOINT",
-			"OAUTH_TOKEN_ENDPOINT",
-			"OAUTH_ALLOWED_AUTHORIZATION_ENDPOINTS",
-			"OAUTH_ALLOWED_TOKEN_ENDPOINTS":
+		if slices.Contains(OauthIntegrationForPartnerApplicationsPropertiesNames, property.Name) {
 			securityIntegrationProperties[strings.ToLower(property.Name)] = []map[string]any{SecurityIntegrationPropertyToSchema(&property)}
-		default:
-			log.Printf("[WARN] unexpected property %v returned from Snowflake", property.Name)
+		} else {
+			log.Printf("[WARN] unexpected property %v in oauth security integration returned from Snowflake", property.Name)
```
Comment (sfc-gh-jcieslak) [inline pkg/schemas/oauth_security_integration_for_partner_applications.go:63]: maybe just for clarity add a type (partner application / custom clients) here and in the other oauth integration, so we'll know exactly where we "failed".

**Review by sfc-gh-jcieslak — APPROVED** [review]

File: pkg/schemas/security_integration.go (hunk 1 above)
Comment (sfc-gh-asawicki) [reply pkg/schemas/security_integration.go:36]: Yup, let's leave this default with WARN pls

File: pkg/datasources/security_integrations_acceptance_test.go [hunk 3]
```diff
@@ -8,24 +8,521 @@ import (
 	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
 
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/helpers/random"
-	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/testenvs"
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
 	"github.com/hashicorp/terraform-plugin-testing/config"
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
 	"github.com/hashicorp/terraform-plugin-testing/tfversion"
 )
```
Comment (sfc-gh-asawicki) [inline pkg/datasources/security_integrations_acceptance_test.go:17]: can we add one more test (with not that many assertions) that shows that it works when multiple different type integrations are created, pls?

**Review by sfc-gh-asawicki — APPROVED** [review]

File: pkg/schemas/oauth_security_integration_for_partner_applications.go (hunk 2 above)
Comment (sfc-gh-jmichalak) [reply pkg/schemas/oauth_security_integration_for_partner_applications.go]: Done.

File: pkg/schemas/security_integration.go (hunk 1 above)
Comment (sfc-gh-jmichalak) [reply pkg/schemas/security_integration.go:39]: Added.

File: pkg/datasources/security_integrations_acceptance_test.go (hunk 3 above)
Comment (sfc-gh-jmichalak) [reply pkg/datasources/security_integrations_acceptance_test.go:18]: Done. I couldn't find a way to assert that output is sorted, but it seems like snowflake show output is always sorted by integration type. I tried this a few times locally and it looks like this order should be consistent.

Comment (sfc-gh-asawicki) [reply pkg/datasources/security_integrations_acceptance_test.go:18]: We will observe and adjust if needed.

**Review by sfc-gh-asawicki — APPROVED** [review]

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**